File: libraries/typescript/src/reader.ts
import { IfcxDocument } from './document.js';
import type { IfcxDocument as IIfcxDocument } from './types.js';

/**
 * Reads IFCX (JSON) files
 */
export class IfcxReader {
  /** Read from JSON string */
  static fromString(json: string): IfcxDocument {
    const data = JSON.parse(json) as IIfcxDocument;
    return IfcxDocument.fromJSON(data);
  }

  /** Read from buffer (UTF-8 encoded JSON) */
  static fromBuffer(buffer: Uint8Array): IfcxDocument {
    const decoder = new TextDecoder('utf-8');
    const json = decoder.decode(buffer);
    return IfcxReader.fromString(json);
  }
}

File: libraries/python/ifcx/reader.py
"""IFCX file reader."""

from __future__ import annotations

import json
from pathlib import Path

from ifcx.document import IfcxDocument


class IfcxReader:
    """Reads IFCX (JSON) files."""

    @staticmethod
    def from_file(path: str | Path) -> IfcxDocument:
        """Read IFCX from file."""
        path = Path(path)
        text = path.read_text(encoding="utf-8")
        return IfcxDocument.from_json(text)

    @staticmethod
    def from_string(json_str: str) -> IfcxDocument:
        """Read IFCX from JSON string."""
        return IfcxDocument.from_json(json_str)

    @staticmethod
    def from_bytes(data: bytes) -> IfcxDocument:
        """Read IFCX from bytes."""
        return IfcxDocument.from_json(data.decode("utf-8"))
